taqueria-vscode-extension/src/lib/gui/helpers/GrouppedFileItem.ts

```diff
@@ -0,0 +1,66 @@
+import path from 'path';
+import * as vscode from 'vscode';
+
+export class GrouppedFileItem {
+	readonly relativePath: string;
+	readonly isChild: boolean;
+	readonly expectedParentName: string | undefined;
+
+	constructor(public readonly uri: vscode.Uri, folder: string, patterns: RegExp[]) {
+		this.relativePath = path.relative(folder, uri.path);
+		const result = GrouppedFileItem.getFirstMatch(this.relativePath, patterns);
+		if (result) {
+			this.expectedParentName = `${result.groups?.name}${result.groups?.ext}`;
+			this.isChild = true;
+		} else {
+			this.isChild = false;
+			this.expectedParentName = undefined;
+		}
+	}
+
+	children: GrouppedFileItem[] = [];
+
+	toString() {
+		return this.uri.toString();
+	}
+
+	private static contractChildrenSpecifiers = [/(?<name>.+)\.(?:parameter|storage)List(?<ext>\.(?:m|re|js)?ligo)/];
+	private static artifactChildrenSpecifiers = [
+		/(?<name>.+)\.default_storage(?<ext>\.tz)/,
+		/(?<name>.+)\.(?:storage|parameter|expression)\..*(?<ext>\.tz)/,
```

Comment:
Instead of `.*`, `.+` seems a little more accurate right? Because something always exists between the dots. E.g. `counter.storage.someNonEmptyWord.tz`

Reply:
Nice catch, yes.